packages/superset-ui-preset-chart-xy/src/utils/createTickComponent.tsx

```diff
@@ -16,28 +20,26 @@ export default function createTickComponent({
     }
     const yOffset = orientation === 'top' ? -3 : 0;
 
-    const propTypes = {
-      dy: PropTypes.number,
-      formattedValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-      x: PropTypes.number.isRequired,
-      y: PropTypes.number.isRequired,
-    };
-    const defaultProps = {
-      dy: null,
-      formattedValue: '',
-    };
-
-    const TickComponent = ({ x, y, dy, formattedValue, ...textStyle }) => (
+    const TickComponent = ({
```

Comment:
I realize this component wasn't added here, but just to understand why we didn't leverage the `@vx/text` component for auto wrapping / easy rotation etc., is it mostly that the manual calculations are needed to move the margins dynamically?